diff --git a/integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/ContentPresentation.java b/integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/ContentPresentation.java
index 7ba2dae..f089bb3 100644
--- a/integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/ContentPresentation.java
+++ b/integ-ecms/integ-ecms-social/src/main/java/org/exoplatform/wcm/ext/component/activity/ContentPresentation.java
@@ -88,7 +88,7 @@ public class ContentPresentation extends UIBaseNodePresentation {
     String ext = mimeTypeSolver.getExtension(mimeType);
     String fileName = node.getName();
     if (fileName.lastIndexOf("." + ext) < 0) {
-      fileName = fileName + "." + ext;
+      fileName = new StringBuffer(fileName).append(".").append(ext).toString();
     }
     dresource.setDownloadName(fileName);
     return dservice.getDownloadLink(dservice.addDownloadResource(dresource));
diff --git a/integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/impl/ForumUIActivity.java b/integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/impl/ForumUIActivity.java
index ec35e91..4140d35 100644
--- a/integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/impl/ForumUIActivity.java
+++ b/integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/impl/ForumUIActivity.java
@@ -31,11 +31,13 @@ public class ForumUIActivity extends BaseKSActivity {
   @SuppressWarnings("unused")
   private String getReplyLink() {
     String link = getViewLink();
-    if (!link.endsWith("/"))
-      link += "/";
+    StringBuffer sb = new StringBuffer(getViewLink());
+    if (sb.lastIndexOf("/") == -1 || sb.lastIndexOf("/") != sb.length() - 1) {
+      sb.append("/");
+    }
     // add signal to show reply form
-    link += "false";
-    return link;
+    sb.append("false");
+    return sb.toString();
   }
 
   private String getViewLink() {
diff --git a/integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java b/integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java
index 987ddf0..07978ac 100644
--- a/integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java
+++ b/integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java
@@ -92,7 +92,7 @@ public class WikiSpaceActivityPublisher extends PageWikiListener {
       templateParams.put(VIEW_CHANGE_URL_KEY, Utils.getURL(page.getURL(), verName));
       excerpt = page.getComment();
     }
-    excerpt = (excerpt.length() > EXCERPT_LENGTH) ? excerpt.substring(0, EXCERPT_LENGTH) + "..." : excerpt;
+    excerpt = (excerpt.length() > EXCERPT_LENGTH) ? new StringBuffer(excerpt.substring(0, EXCERPT_LENGTH)).append("..." ).toString():excerpt;
     templateParams.put(PAGE_EXCERPT, excerpt);
     templateParams.put(org.exoplatform.social.core.BaseActivityProcessorPlugin.TEMPLATE_PARAM_TO_PROCESS, PAGE_EXCERPT);
     activity.setTemplateParams(templateParams);
